Phase One Project Submission

Please fill out:

Student name: Josephine Gathenya

Student pace: part time

Scheduled project review date/time:

Instructor name:

Blog post URL:

PROJECT STATEMENT

Microsoft Company would wish to join the creative movie industry by creating a new movie studio. They however do not have a background in creating movies and, would like to know, which films are performing the best at the Box Office.

The objectives of this project are to use data analytic tools and techniques to gain useful insights and decision making that will help Microsoft to decide what type of films to create.



FILES

I will be working with the following files:

imdb.title.basics

imdb.title.ratings

bom.movie_gross

In [2]:
ls

 Volume in drive C has no label.
 Volume Serial Number is CEA5-6E23

 Directory of C:\Users\lenovo\Desktop\Flatiron\Phase 1 project

02/11/2023  22:11    <DIR>          .
02/11/2023  22:11    <DIR>          ..
02/11/2023  22:11    <DIR>          .ipynb_checkpoints
02/11/2023  21:00           142,555 bom.movie_gross.csv
02/11/2023  22:10             4,021 phase 1 project.ipynb
02/11/2023  22:11             1,688 Phase one project.ipynb
02/11/2023  21:01        11,852,240 title.basics.csv
02/11/2023  21:01         1,950,137 title.ratings.csv
               5 File(s)     13,950,641 bytes
               3 Dir(s)  145,653,567,488 bytes free


In [3]:
# importing the needed libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [4]:
#parsing csv/tsv files to pandas dataframe
title_basics = pd.read_csv('title.basics.csv')

In [5]:
title_ratings = pd.read_csv('title.ratings.csv')

In [6]:
bom_movie_gross = pd.read_csv('bom.movie_gross.csv')

In [10]:
#exploring the file data of the title.basics.csv
#first rows
title_basics.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [13]:
#rows and columns
title_basics.shape

(146144, 6)

In [27]:
title_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


In [14]:
#summary of the all the data 
title_basics.describe

<bound method NDFrame.describe of            tconst                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ash

In [15]:
#column names
title_basics.columns

Index(['tconst', 'primary_title', 'original_title', 'start_year',
       'runtime_minutes', 'genres'],
      dtype='object')

**There are 6 columns in this dataset**

In [16]:
#data types
title_basics.dtypes

tconst              object
primary_title       object
original_title      object
start_year           int64
runtime_minutes    float64
genres              object
dtype: object

In [17]:
#finding any missing values
title_basics.isna().sum()

tconst                 0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

   **There are missing values in the title_basics dataset and is as follows:**

original_title        21

runtime_minutes    31739

genres              5408

**what to do with the missing values?**

In [48]:
#finding unique values
title_basics['original_title'].unique()

array(['Sunghursh', 'Ashad Ka Ek Din', 'The Other Side of the Wind', ...,
       'Dankyavar Danka', '6 Gunn', 'Chico Albuquerque - Revelações'],
      dtype=object)

In [49]:
len(title_basics['original_title'].unique())

137774


**There are 137774 / 146144 unique values in the 'original_title'**

In [46]:
#finding duplicated values
duplicate_values = title_basics[title_basics.duplicated(subset='original_title', keep=False)]
duplicate_values

tconst                     primary_title  \
6       tt0112502                           Bigfoot   
19      tt0247643  Los pájaros se van con la muerte   
21      tt0250404                         Godfather   
30      tt0285252                    Life's a Beach   
34      tt0297400                         Snowblind   
...           ...                               ...   
146117  tt9913418                         Fragments   
146119  tt9913594                       Bacchanalia   
146120  tt9913936                          Paradise   
146125  tt9914642                         Albatross   
146134  tt9916160                        Drømmeland   

                          original_title  start_year  runtime_minutes  \
6                                Bigfoot        2017              NaN   
19      Los pájaros se van con la muerte        2011            110.0   
21                             Godfather        2012              NaN   
30                        Life's a Beach        2012            100.0   
34                             Snowblind        2015              NaN   
...                                  ...         ...              ...   
146117                         Fragments        2019              NaN   
146119                       Bacchanalia        2017             72.0   
146120                          Paradise        2019              NaN   
146125                         Albatross        2017              NaN   
146134                        Drømmeland        2019             72.0   

                        genres  
6              Horror,Thriller  
19               Drama,Mystery  
21                 Crime,Drama  
30                      Comedy  
34                 Crime,Drama  
...                        ...  
146117    Drama,Mystery,Sci-Fi  
146119  Drama,Mystery,Thriller  
146120             Crime,Drama  
146125             Documentary  
146134             Documentary  

[13751 rows x 6 columns]



there are 13751/146144 rows where values in the 'original_title' column are the same as the 'primary_title' column


In [11]:
title_ratings.head()

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [18]:
title_ratings.shape

(73856, 3)

In [26]:
title_ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   tconst         73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB


In [19]:
title_ratings.describe

<bound method NDFrame.describe of            tconst  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]>

In [20]:
title_ratings.columns

Index(['tconst', 'averagerating', 'numvotes'], dtype='object')

**There are 3 columns in this dataset**

In [21]:
title_ratings.dtypes

tconst            object
averagerating    float64
numvotes           int64
dtype: object

In [22]:
title_ratings.isna().sum()

tconst           0
averagerating    0
numvotes         0
dtype: int64

**There are no missing values in the title_ratings dataset**

In [12]:
bom_movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [23]:
bom_movie_gross.shape

(3387, 5)

In [25]:
bom_movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [28]:
bom_movie_gross.describe

<bound method NDFrame.describe of                                             title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700

In [29]:
bom_movie_gross.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

**There are 5 columns in this dataset**

In [30]:
bom_movie_gross.dtypes

title              object
studio             object
domestic_gross    float64
foreign_gross      object
year                int64
dtype: object

In [31]:
bom_movie_gross.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

**There are missing values in 3 of 5 columns of the bom_movie_gross dataset and is as follows:**

studio               5

domestic_gross      28

foreign_gross     1350
